dotNetInstallerToolsLib/OsUtil.h

```diff
@@ -174,6 +174,35 @@ namespace DVLib
 	bool IsOperatingSystemID(OperatingSystem os, const std::wstring& os_filter);
 	bool IsInOperatingSystemInRange(OperatingSystem os, const std::wstring& os_filter, 
 		OperatingSystem l, OperatingSystem r);
+
+	// operating system product type
+
+	enum OperatingSystemProductType
+	{
+		osNone = -1,
+
+		Workstation = 1,
+		DomainController = 2,
+		Server = 3
+	};
+
+	OperatingSystemProductType GetOperatingSystemProductType();
+	bool IsOperatingSystemProductType(BYTE osProductType_in, const std::wstring& filter);
+	WORD wstring2osType(const std::wstring& osProductType);
+
+	struct OPERATING_SYSTEM_PRODUCT_TYPE
+	{
+		LPCWSTR name;
+		BYTE productType;
+	};
+
+	static const OPERATING_SYSTEM_PRODUCT_TYPE operating_system_product_types[] =
+	{
+		{ L"workstation", VER_NT_WORKSTATION },
+		{ L"server", VER_NT_SERVER },
+		{ L"domainController", VER_NT_DOMAIN_CONTROLLER },

```

Comment:
Do we have other uses of camelCase? If not, maybe just "dc" or "domaincontroller" in 1-word and all lowercase?

Reply:
I tried to stay consistent and use the same case as the [identifiers for the different Windows operating systems](https://github.com/dotnetinstaller/dotnetinstaller/blob/a7d0c5990805e9c5c4017433233bdaf9aba01a3e/dotNetInstallerToolsLib/OsUtil.h#L110):
```
winXP
winServer2003
winServer2008
winVista
winServer2019
winServer2022
```

Would you still like me to use "dc" instead of "domainController"?

Thank you.

Reply:
Oh I see. Let's keep it with this.

Reply:
Thank you.